R/teal_slice.R

```diff
@@ -308,245 +205,28 @@ format.teal_slice <- function(x, show_all = FALSE, trim_lines = TRUE, ...) {
-                  arg = name_ii,
-                  varname = name_iii
-                )
-                slices <- c(slices, list(as.teal_slice(args)))
-              }
-            }
-          }
-        }
-      }
-    }
-  }
-
-  if (length(slices) == 0L && length(x) != 0L) {
-    stop("conversion to filter_slices failed")
-  }
-
-  do.call(filter_settings, c(slices, list(include_varnames = attr(x, "filterable"))))
-}
-
-
-# subset method for `teal_slices`
-#' @export
-#' @rdname teal_slice
-#' @keywords internal
-#'
-`[.teal_slices` <- function(x, i) {
-  if (missing(i)) i <- seq_along(x)
-  if (length(i) == 0L) {
-    return(x[0])
-  }
-  if (is.logical(i) && length(i) > length(x)) stop("subscript out of bounds")
-  if (is.numeric(i) && max(i) > length(x)) stop("subscript out of bounds")
-  if (is.character(i)) {
-    if (!all(is.element(i, names(x)))) stop("subscript out of bounds")
-    i <- which(is.element(i, names(x)))
-  }
-
-  y <- NextMethod("[")
-  attrs <- attributes(x)
-  attrs$names <- attrs$names[i]
-  datanames <- unique(unlist(vapply(y, function(ts) shiny::isolate(ts[["dataname"]]), character(1L))))
-  attrs[["exclude_varnames"]] <- Filter(Negate(is.null), attr(x, "exclude_varnames")[datanames])
-  attrs[["include_varnames"]] <- Filter(Negate(is.null), attr(x, "include_varnames")[datanames])
-  attributes(y) <- attrs
-  y
-}
-
-
-# concatenate method for `teal_slices`
-#' @export
-#' @rdname teal_slice
-#' @keywords internal
-#'
-c.teal_slices <- function(...) {
-  x <- list(...)
-  checkmate::assert_true(all(vapply(x, is.teal_slices, logical(1L))), .var.name = "all arguments are teal_slices")
-
-  excludes <- lapply(x, attr, "exclude_varnames")
-  names(excludes) <- NULL
-  excludes <- unlist(excludes, recursive = FALSE)
-  excludes <- excludes[!duplicated(names(excludes))]
-
-  includes <- lapply(x, attr, "include_varnames")
-  names(includes) <- NULL
-  includes <- unlist(includes, recursive = FALSE)
-  includes <- includes[!duplicated(names(includes))]
-
-  count_types <- lapply(x, attr, "count_type")
-  count_types <- unique(unlist(count_types))
-
-  do.call(
-    filter_settings,
-    c(
-      unique(unlist(x, recursive = FALSE)),
-      list(
-        include_varnames = if (length(includes)) includes,
-        exclude_varnames = if (length(excludes)) excludes,
-        count_type = count_types
-      )
-    )
-  )
-}
-
-#' @export
-#' @rdname teal_slice
-#' @keywords internal
-#'
-format.teal_slices <- function(x, show_all = FALSE, trim_lines = TRUE, ...) {
-  checkmate::assert_flag(show_all)
-  checkmate::assert_flag(trim_lines)
-
-  slices_list <- slices_to_list(x)
-
-  if (!show_all) slices_list$slices <- lapply(slices_list$slices, function(slice) Filter(Negate(is.null), slice))
-
-  jsonify(slices_list, trim_lines)
-}
-
-#' @export
-#' @rdname teal_slice
-#' @keywords internal
-#'
-print.teal_slices <- function(x, ...) {
-  cat(format(x, ...), "\n")
-}
-
-
-# helpers ----
+# format utils -----
```

Comment:
I agree but it will give us another reason for unnecessary discussions. Instead of having `@keywords internal` we will have to decide if function suits for `@keywords internal` or `@noRd`. We have no `@noRd` in this package and I'm happy to keep it this way